python/dgl/distributed/kvstore.py

```diff
@@ -0,0 +1,1072 @@
+        self.name = state
+
+    def process_request(self, server_state):
+        kv_store = server_state.kv_store
+        if kv_store.data_store.__contains__(self.name) is False:
+            raise RuntimeError("KVServer Cannot find data tensor with name: %s" % self.name)
+        data_shape = F.shape(kv_store.data_store[self.name])
+        res = GetPartShapeResponse(data_shape)
+        return res
+
+SEND_META_TO_BACKUP = 901239
+SEND_META_TO_BACKUP_MSG = "Send_Meta_TO_Backup"
+
+class SendMetaToBackupResponse(rpc.Response):
+    """Send a confirmation signal (just a short string message)
+    of SendMetaToBackupRequest to client.
+    """
+    def __init__(self, msg):
+        self.msg = msg
+
+    def __getstate__(self):
+        return self.msg
+
+    def __setstate__(self, state):
+        self.msg = state
+
+class SendMetaToBackupRequest(rpc.Request):
+    """Send meta data to backup server and backup server
+    will use this meta data to read shared-memory tensor.
+
+    Parameters
+    ----------
+    name : str
+        data name
+    dtype : str
+        data type string
+    shape : tuple of int
+        data shape
+    policy_str : str
+        partition-policy string, e.g., 'edge' or 'node'.
+    """
+    def __init__(self, name, dtype, shape, policy_str):
+        self.name = name
+        self.dtype = dtype
+        self.shape = shape
+        self.policy_str = policy_str
+
+    def __getstate__(self):
+        return self.name, self.dtype, self.shape, self.policy_str
+
+    def __setstate__(self, state):
+        self.name, self.dtype, self.shape, self.policy_str = state
+
+    def process_request(self, server_state):
+        kv_store = server_state.kv_store
+        assert kv_store.is_backup_server()
+        shared_data = empty_shared_mem(self.name+'-kvdata-', False, self.shape, self.dtype)
+        dlpack = shared_data.to_dlpack()
+        kv_store.data_store[self.name] = F.zerocopy_from_dlpack(dlpack)
+        for _, policy in kv_store.part_policy.items():
+            if policy.policy_str == self.policy_str:
+                kv_store.part_policy[self.name] = policy
+                res = SendMetaToBackupResponse(SEND_META_TO_BACKUP_MSG)
+                return res
+        raise RuntimeError("Cannot find any partition policy match \
+            the requested policy : %s" % self.policy_str)
+
+############################ KVServer ###############################
+
+def default_push_handler(target, name, id_tensor, data_tensor):
+    """Default handler for PUSH message.
+
+    On default, _push_handler perform scatter_row() operation for the tensor.
+
+    Parameters
+    ----------
+    target : tensor
+        target tensor
+    name : str
+        data name
+    id_tensor : tensor
+        a vector storing the ID list.
+    data_tensor : tensor
+        a tensor with the same row size of id
+    """
+    # TODO(chao): support Tensorflow backend
+    target[name][id_tensor] = data_tensor
```

Comment:
how to support tensorflow backend?

Reply:
Not supported for now. Let's come back later after we fully understand what kind functions are needed here.